### PR TITLE
SAA-410: Add recurrence to create appointment

### PR DESCRIPTION
**Pull request by iterative-it**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/AppointmentRepeatPeriodTest.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.junit.jupiter.api.Test
+import java.time.DayOfWeek
+import java.time.LocalDate
+
+class AppointmentRepeatPeriodTest {
+  @Test
+  fun `weekday next date increments by one day`() {
+    val monday = LocalDate.of(2023, 3, 13)
+    val nextDate = AppointmentRepeatPeriod.WEEKDAY.nextDate(monday)
+    Assertions.assertThat(nextDate.dayOfMonth).isEqualTo(14)
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/AppointmentRepeatPeriodTest.kt:13]: Remove `Assertions` from the assertions?  That classic copy and paste from Intellij perhaps?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/AppointmentRepeatPeriodTest.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    Assertions.assertThat(nextDate.dayOfWeek).isEqualTo(DayOfWeek.TUESDAY)
+  }
+
+  @Test
+  fun `weekday next date increments friday to monday`() {
+    val friday = LocalDate.of(2023, 3, 17)
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/AppointmentRepeatPeriodTest.kt:19]: Perhaps pull the monday and fridays out into class level vals?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentServiceTest.kt [hunk 3]
```diff
@@ -303,4 +306,33 @@ class AppointmentServiceTest {
...
+            PrisonerSearchPrisonerFixture.instance(prisonerNumber = request.prisonerNumbers.first(), bookingId = 1, prisonId = request.prisonCode!!),
+          ),
+        ),
+      )
+    whenever(appointmentRepository.saveAndFlush(appointmentEntityCaptor.capture())).thenReturn(appointmentEntity(repeatPeriod = AppointmentRepeatPeriod.WEEKLY, numberOfOccurrences = 3))
+
+    service.createAppointment(request, principal)
+
+    with(appointmentEntityCaptor.value) {
+      with(occurrences()) {
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentServiceTest.kt:333]: Given there is only one assertion perhaps just change to assertThat(occurances).hasSize(3) ? Meh.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/Appointment.kt [hunk 4]
```diff
@@ -32,9 +33,9 @@ import uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.model.Appointme
 data class Appointment(
   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
-  val appointmentId: Long = -1,
+  val appointmentId: Long = 0,
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/Appointment.kt:36]: What was the upshot of changing these to zero's in the end?

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/AppointmentRepeatPeriodTest.kt (hunk 1 above)
Comment (iterative-it) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/AppointmentRepeatPeriodTest.kt]: Think I've remove all these now...

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/AppointmentRepeatPeriodTest.kt (hunk 2 above)
Comment (iterative-it) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/AppointmentRepeatPeriodTest.kt]: Good idea. Done

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentServiceTest.kt (hunk 3 above)
Comment (iterative-it) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentServiceTest.kt]: Done

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Pull request merged**